experimenter/experimenter/jetstream/models.py

```diff
@@ -329,71 +334,71 @@ def append_conversion_count(self, primary_metrics_set):
 """
 
 
+# XXX: does this need @lru_cache ?
 def create_results_object_model(data: JetstreamData):
-    branches = {}
-    metrics = {}
-
-    for jetstream_data_point in data:
-        branches[jetstream_data_point.branch] = {}
+    branches = [data_point.branch for data_point in data]
 
     # create a dynamic model with all branches, leveraging BranchComparisonData
-    branches_data = {b: BranchComparisonData() for b in branches}
     PairwiseBranchComparisonData = create_model(
         "PairwiseBranchComparisonData",
-        **branches_data,
+        **{
+            branch: (BranchComparisonData, Field(default_factory=BranchComparisonData))
+            for branch in branches
+        }
     )
 
     # create a dynamic model with all branches, leveraging SignificanceData
-    branches_significance_data = {b: SignificanceData() for b in branches}
     PairwiseSignificanceData = create_model(
         "PairwiseSignificanceData",
-        **branches_significance_data,
+        **{
+            branch: (SignificanceData, Field(default_factory=SignificanceData))
+            for branch in branches
+        }
     )
 
     class PairwiseMetricData(MetricData):
-        difference: PairwiseBranchComparisonData
-        relative_uplift: PairwiseBranchComparisonData
-        significance: PairwiseSignificanceData
-
-    for jetstream_data_point in data:
-        metrics[jetstream_data_point.metric] = PairwiseMetricData(
-            absolute=BranchComparisonData(),
-            difference=PairwiseBranchComparisonData(),
-            relative_uplift=PairwiseBranchComparisonData(),
-            significance=PairwiseSignificanceData(),
-        )
-
-    search_data = {k: v for k, v in metrics.items() if k in SEARCH_METRICS}
-    usage_data = {k: v for k, v in metrics.items() if k in USAGE_METRICS}
-    other_data = {
-        k: v for k, v in metrics.items() if k not in SEARCH_METRICS + USAGE_METRICS
-    }
+        difference: PairwiseBranchComparisonData = Field(default_factory=PairwiseBranchComparisonData)
+        relative_uplift: PairwiseBranchComparisonData = Field(default_factory=PairwiseBranchComparisonData)
+        significance: PairwiseSignificanceData = Field(default_factory=PairwiseSignificanceData)
+
+    metrics = [data_point.metric for data_point in data]
 
     # Dynamically create our grouped models which are dependent on metrics
     # available for a given experiment
-    SearchData = create_model("SearchData", **search_data)
-    UsageData = create_model("UsageData", **usage_data)
-    OtherData = create_model("OtherData", **other_data)
+    SearchData = create_model("SearchData", **{
+        metric_name: (PairwiseMetricData, Field(default_factory=PairwiseMetricData))
+        for metric_name in metrics
+        if metric_name in SEARCH_METRICS
+    })
```

Comment:
I think these are missing the `absolute=BranchComparisonData()` part? (from old line 360)

Reply:
nope, it has a default_factory in the MetricsData class

Reply:
Ah I see, thanks!